proj.pc.in missing CURL and TIFF in Libs.private

Trying to build `libgeotiff` with a static proj7 fails like so:

```
checking for PROJ >= 6 library... checking for proj_create_from_wkt in -lproj... no
checking for internal_proj_create_from_wkt in -lproj... no
checking for internal_proj_create_from_wkt in -linternalproj... no
configure: error: PROJ 6 symbols not found
```

The real problem is that `proj.pc` is missing linker flags for libcurl and libttiff in `Libs.private`:
https://github.com/OSGeo/PROJ/blob/0c5802e0f840cf289dffa02571e414b276bb7d62/proj.pc.in#L12-L13

--- a/proj.pc.in
+++ b/proj.pc.in
@@ -10,5 +10,5 @@ Description: Coordinate transformation software library
 Requires:
 Version: @VERSION@
 Libs: -L${libdir} -lproj
-Libs.private: @SQLITE3_LIBS@ -lstdc++
+Libs.private: @SQLITE3_LIBS@ @TIFF_LIBS@ @CURL_LIBS@ -lstdc++
 Cflags: -I${includedir}

